pkg/router/template/template_helper.go

```diff
@@ -170,6 +173,25 @@ func endpointsForAlias(alias ServiceAliasConfig, svc ServiceUnit) []Endpoint {
 	return endpoints
 }
 
+// sortedMapData returns the sorted data in a haproxy map. The returned data is
+// in alphabetically reverse order. The sortSubGroups option sorts the
+// non-wildcard paths first and adds the sorted wildcard paths at the end.
+// Note: The reversed sorting ensures we do a longest path match first.
+func sortedMapData(name string, sortSubGroups bool) []string {
+	lines, err := fileutil.ReadLines(name)
```

Comment:
Reading a file from within a helper is dangerous, this means a template can read arbitrary files off disk (not possible before), which could lead to a future security hole and allow access to certificate data or other data like the service account key.  I’d prefer not to allow any template to read from disk - it’s just one less escape vector for the template engine.  An author could trivially use this to read the router credentials, and in combination with a future vulnerability could be very dangerous.

Reply:
I do understand those consequences but that said, one doesn't really to read that cert/key info from disk ... all that config is available inside the template in `.State`. 

But yeah not doing a file read/passing a name is definitely better  ... so, based on the above comment (aka just doing a simple `range` and write in the template and computing the info inside different helper functions) - if that works for you, I can create a new PR next week  ... got to pay my dues to the other `devil` this weekend!